library/core/src/any.rs

```diff
@@ -594,10 +670,24 @@ impl dyn Any + Send + Sync {
 /// While `TypeId` implements `Hash`, `PartialOrd`, and `Ord`, it is worth
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(not(bootstrap), lang = "TypeId")]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct TypeId {
-    t: u64,
+    hash: u64,
+
+    // NOTE(eddyb) this follows `hash`, so that the auto-derived `PartialEq`
+    // doesn't waste time comparing `mangling`, when `hash` differs.
+    #[cfg(not(bootstrap))]
+    mangling: &'static TypeManglingStr,
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl fmt::Debug for TypeId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // FIXME(eddyb) perhaps attempt demangling `mangling`.
+        f.debug_struct("TypeId").finish_non_exhaustive()
```

Comment:
This is a debug format regression. Previously it was possible to tell different `TypeId` apart.

Reply:
Hmm, weird, I don't remember doing this but I guess I noticed `TypeId` was infoleaking and reacted by doing this? Ideally we would print the mangling but I'm not sure we should commit to that without a more involved proposal.